index.bs

```diff
@@ -314,6 +354,49 @@ The <dfn method for=FileSystemFileHandle>createWritable(|options|)</dfn> method,
 
 </div>
 
+### The {{FileSystemFileHandle/createSyncAccessHandle()}} method ### {#api-filesystemfilehandle-createsyncaccesshandle}
+
+<div class="note domintro">
+  : |handle| = await |fileHandle| . {{FileSystemFileHandle/createSyncAccessHandle()|createSyncAccessHandle}}()
+  :: Returns a {{FileSystemSyncAccessHandle}} that can be used to read/write from/to the file.
+     Changes made through |handle| might be immediately reflected in the file represented by |fileHandle|.
+     To ensure the changes are reflected in this file, the handle must be flushed or closed.
+
+     Creating a {{FileSystemSyncAccessHandle}} [=file entry/lock/take|takes an exclusive lock=] on the
+     [=FileSystemHandle/entry=] associated with |fileHandle|. This prevents the creation of
+     further {{FileSystemSyncAccessHandle}}s or {{FileSystemWritableFileStream}}s
+     for the entry, until the access handle is closed.
+
+     The returned {{FileSystemSyncAccessHandle}} offers synchronous {{FileSystemSyncAccessHandle/read()}} and
+     {{FileSystemSyncAccessHandle/write()}} methods. This allows for higher performance for critical methods on
+     contexts where asynchronous operations come with high overhead, e.g., WebAssembly.
+
+     For the time being, this method will only succeed when the |fileHandle| belongs to the
+     [=origin private file system=].
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemFileHandle>createSyncAccessHandle()</dfn> method, when invoked, must run these steps:
+
+1. Let |result| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |access| be the result of running [=this=]'s [=FileSystemHandle/entry=]'s
+     [=entry/request access=] given "`readwrite`".
+     If that throws an exception, [=reject=] |result| with that exception and abort.
+  1. If |access| is not "{{PermissionState/granted}}",
+     reject |result| with a {{NotAllowedError}} and abort.
```

Comment:
This is a comment that applies throughout: while you're "in parallel" you first need to queue a task and from that task you can manipulate state in the other thread, such as rejecting or resolving a promise.

You want to invoke HTML's "queue a global task" for this. (The Storage Standard hasn't been updated yet to account for this.) I suspect we want a task source that most storage APIs can share, e.g., "storage task source", though we could have one dedicated to this API I suppose, e.g., "file system task source". If we go with the former we'd need to add it to the Storage Standard. I can help with that.